bundles/org.openhab.binding.velux/src/main/java/org/openhab/binding/velux/internal/handler/VeluxBindingHandler.java

```diff
@@ -82,9 +84,16 @@ public VeluxBindingHandler(Thing thing, final Localization localization) {
      * @param channelUID for type {@link ChannelUID}.
      * @return thingTypeUID of type {@link ThingTypeUID}.
      */
-    @SuppressWarnings("deprecation")
     private ThingTypeUID thingTypeUIDOf(ChannelUID channelUID) {
-        return channelUID.getThingUID().getThingTypeUID();
+        Channel chan = getThing().getChannel(channelUID);
+        if (chan != null) {
+            ChannelTypeUID type = chan.getChannelTypeUID();
+            if (type != null) {
+                return new ThingTypeUID(type.getBindingId(), type.getId());
+            }
+        }
+        logger.warn("thingTypeUIDOf({}) failed.", channelUID);
+        return new ThingTypeUID(VeluxBindingConstants.BINDING_ID, "???");
```

Comment:
would suggest to replace the three questionmarks by a global constant.

Reply:
Thank you. Suggestion adopted.